### ISSUE TITLE
`papis add` fails on files with spaces when smart matching is active

### ISSUE THREAD

**Issue by stuartthomas25** [papis/papis#333]
Papis is not able to add pdf files with spaces.

* papis version: papis, version 0.11.1

# how to reproduce the issue
Output of `papis --log INFO add kwon\ yakovenko.pdf`:
```
Traceback (most recent call last):
  File "/opt/miniforge3/bin/papis", line 8, in <module>
    sys.exit(run())
  File "/opt/miniforge3/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/opt/miniforge3/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/opt/miniforge3/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/miniforge3/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/miniforge3/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/opt/miniforge3/lib/python3.9/site-packages/papis/commands/add.py", line 533, in cli
    matching_importers = sum((
  File "/opt/miniforge3/lib/python3.9/site-packages/papis/commands/add.py", line 534, in <genexpr>
    papis.utils.get_matching_importer_or_downloader(f)
  File "/opt/miniforge3/lib/python3.9/site-packages/papis/utils.py", line 233, in get_matching_importer_or_downloader
    importer = importer_cls.match(
  File "/opt/miniforge3/lib/python3.9/site-packages/papis/arxiv.py", line 287, in match
    validate_arxivid(uri)
  File "/opt/miniforge3/lib/python3.9/site-packages/papis/arxiv.py", line 115, in validate_arxivid
    urllib.request.urlopen(request)
  File "/opt/miniforge3/lib/python3.9/urllib/request.py", line 214, in urlopen
    return opener.open(url, data, timeout)
  File "/opt/miniforge3/lib/python3.9/urllib/request.py", line 517, in open
    response = self._open(req, data)
  File "/opt/miniforge3/lib/python3.9/urllib/request.py", line 534, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/opt/miniforge3/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/opt/miniforge3/lib/python3.9/urllib/request.py", line 1389, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/opt/miniforge3/lib/python3.9/urllib/request.py", line 1346, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/opt/miniforge3/lib/python3.9/http/client.py", line 1253, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/opt/miniforge3/lib/python3.9/http/client.py", line 1264, in _send_request
    self.putrequest(method, url, **skips)
  File "/opt/miniforge3/lib/python3.9/http/client.py", line 1098, in putrequest
    self._validate_path(url)
  File "/opt/miniforge3/lib/python3.9/http/client.py", line 1198, in _validate_path
    raise InvalidURL(f"URL can't contain control characters. {url!r} "
http.client.InvalidURL: URL can't contain control characters. '/abs/kwon yakovenko.pdf' (found at least ' ')
```

**Comment by stuartthomas25** [comment]
I believe this issue is due to an uncaught exception. The `validate_arxiv` method checks for URLError, but a InvalidURL error is thrown instead. This is not caught by the try-except block.

This same issue exists for `validate_doi`

**Comment by alejandrogallo** [comment]
hello Stuart, yes this seems to be the case indeed.

The thing about `papis add` as you already probably know
is that it tries to match the input to some possibilities,
it is a url? it is a doi? it is a pdf?
Then by checking if it is a url then it crashes whenever
there are spaces in the name, it's fixable, actually if you do

```
papis add --from doi 132/132 "my file with spaces.pdf"
```
it works because when one specifies `--from` then it does
not do this "smart" matching.

I can take a look to fix this, as it is a bug, if you want to
give it a go whenever you find the time it is also great.

**Comment by alexfikl** [comment]
@stuartthomas25 The nasty error from the raised exception should have been fixed by #485, which uses `requests` and checks errors a bit differently.

Feel free to reopen if it's not completely fixed for you!
